operator/pkg/ciliumidentity/controller_test.go

```diff
@@ -197,3 +204,253 @@ func verifyCIDUsageInCES(ctx context.Context, fakeClient *k8sClient.FakeClientse
 
 	return nil
 }
+
+// TestCiliumIdentityManagement verifies that the CID management system is
+// working correctly when pods are created, updated and delete. Delete is not
+// yet working because the legacy CID GC is still used.
+// https://github.com/cilium/cilium/issues/34740
+func TestCiliumIdentityManagement(t *testing.T) {
```

Comment:
Aren't these tests mostly covered by `TestReconcilePod` from the reconciler_test.go? Though I can see some benefits including the leaked resources.

Reply:
It's true that reconciler is the main part of these tests too, but the reconciler tests don't actually run the controller and watchers, so it can catch additional things here.